src/WorldState/WorldStateClient.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using WorldState.Data.Models;
+
+namespace WorldState
+{
+    public class WorldStateClient : IDisposable
+    {
+        public Platform Platform { get; }
+
+        internal HttpClient HttpClient { get; }
+
+        private JsonSerializer serializer;
+        private UriBuilder     builder;
```

Comment:
Currently, no. It didn't cross my mind on `UriBuilder` because I thought `HttpClient` itself wasn't even thread-safe? It was also something I read a long time ago so I'll look it up again and see what should be done best.

Reply:
[Docs say GetAsync is thread-safe.](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient#remarks)

Reply:
I see. That complicates things.

Reply:
Not really. You can use one of the constructors for `Uri` instead:

```cs
var baseUri = new Uri("api.warframestat.us/pc/");
var subUri = new Uri(baseUri, "blah"); // api.warframestat.us/pc/blah
```